app/Controllers/Donation/AddDonationController.php

```diff
@@ -49,13 +46,11 @@ public function index( FunFunFactory $ffFactory, Request $request ): Response {
 	private function newHttpResponse( AddDonationResponse $responseModel ): Response {
 		// for immediately completed payments like Direct Debit / Banktransfer there is no redirect URL
 		if ( $responseModel->getPaymentProviderRedirectUrl() === '' ) {
-			return new RedirectResponse( $this->ffFactory->getUrlGenerator()->generateAbsoluteUrl(
-				Routes::SHOW_DONATION_CONFIRMATION,
-				[
-					'id' => $responseModel->getDonation()->getId(),
-					'accessToken' => $responseModel->getAccessToken()
-				]
-			) );
+			$authenticationLoader = $this->ffFactory->getUrlAuthenticationLoader();
+			$authenticator = $authenticationLoader->getDonationUrlAuthenticator( $responseModel->getDonation()->getId() );
+			$url = $this->ffFactory->getUrlGenerator()->generateAbsoluteUrl( Routes::SHOW_DONATION_CONFIRMATION );
+			$url = $authenticator->addAuthenticationTokensToApplicationUrl( $url );
+			return new RedirectResponse( $url );
```

Comment:
This logic and branch should be moved into the Payment bounded context so we'll always redirect.